socotratools/socotratools/client.py

```diff
@@ -460,48 +462,57 @@ def get_policy_status(self, policy_locator):
-                           end_exposures=[], action=None):
-
+                           effective_timestamp=None,
+                           field_values=None,
+                           add_field_groups=None,
+                           remove_field_groups=None,
+                           update_field_groups=None,
+                           add_exposures=None,
+                           end_exposures=None,
+                           update_exposures=None,
+                           end_timestamp=None,
+                           # above: optional
+                           state='application'):
         data = {'endorsementName': endorsement_name,
                 'startTimestamp': effective_timestamp,
-                'newPolicyEndTimestamp': end_timestamp,
                 'fieldValues': field_values,
                 'addFieldGroups': add_field_groups,
-                'updateFieldGroups': update_field_groups,
                 'removeFieldGroups': remove_field_groups,
+                'updateFieldGroups': update_field_groups,
                 'addExposures': add_exposures,
-                'updateExposures': update_exposures,
                 'endExposures': end_exposures,
-                'action': action
-                }
-
+                'updateExposures': update_exposures,
+                'newPolicyEndTimestamp': end_timestamp,
+                'state': state}
         return self.__post("/policies/" + policy_locator +
                            "/endorsements", data)
 
     def update_endorsement(self, endorsement_locator,
-                           effective_timestamp=None, end_timestamp=None,
-                           field_values={}, add_field_groups=[],
-                           update_field_groups=[], remove_field_groups=[],
-                           add_exposures=[], update_exposures=[],
-                           end_exposures=[], action=None):
-
-        update = {'fieldValues': field_values,
-                  'addFieldGroups': add_field_groups,
-                  'updateFieldGroups': update_field_groups,
-                  'removeFieldGroups': remove_field_groups,
-                  'addExposures': add_exposures,
-                  'updateExposures': update_exposures,
-                  'endExposures': end_exposures
-                  }
-
-        data = {'endorsementUpdate': {},
-                'action': action
-                }
-
+                           effective_timestamp=None,
+                           field_values=None,
+                           add_field_groups=None,
+                           remove_field_groups=None,
+                           update_field_groups=None,
+                           add_exposures=None,
+                           end_exposures=None,
+                           update_exposures=None,
+                           end_timestamp=None,
+                           # above: optional
+                           action='update'):
```

Comment:
Note: **This `action` type is already supported for both endorsements/renewals today**, even though the docs preview doesn't reflect these facts yet.

I have a small proposal for our consideration: when an update is intended, require the `action='update'` (which the default provided in the client here) so that the API is ergonomic to **prevent unintended updates** prior to accept/issue.

The key change here (also: captured for renewals, below) is that an update (to an endorsement/renewal) will NOT be sent in the request UNLESS there's a "real update" requested to the endorsement/renewal. Effectively, the proposal is to allow only explicit changes to an endorsement/renewal (which capture a change to a policy) rather than the two "implicit" changes that may take place: update-then-accept or update-then-accept/issue.